src/relax/ir/block_builder.cc

```diff
@@ -170,13 +170,13 @@ class BlockBuilderImpl : public BlockBuilderNode {
         auto it = shape_var_map.find(shape_var);
         if (it == shape_var_map.end()) {
           shape_var_map.Set(shape_var, shape_expr);
+          analyzer_.MarkGlobalNonNegValue(shape_var);
```

Comment:
Might be worth commenting that indicating this results in much simpler symbolic shapes in many cases.